operators/pkg/instance-creation/exposition_test.go

```diff
@@ -0,0 +1,174 @@
+package instance_creation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func TestForgeService(t *testing.T) {
+	var (
+		name      = "usertest"
+		namespace = "namespacetest"
+	)
+
+	ownerRef := []metav1.OwnerReference{{
+		APIVersion: "crownlabs.polito.it/v1alpha2",
+		Kind:       "Instance",
+		Name:       "Test1",
+	},
+	}
+
+	service := ForgeService(name, namespace, ownerRef)
+
+	assert.Equal(t, service.ObjectMeta.Name, name+"-svc")
+	assert.Equal(t, service.ObjectMeta.Namespace, namespace)
+	assert.Equal(t, service.OwnerReferences[0].APIVersion, "crownlabs.polito.it/v1alpha2")
+	assert.Equal(t, service.OwnerReferences[0].Kind, "Instance")
+	assert.Equal(t, service.OwnerReferences[0].Name, "Test1")
```

Comment:
It should also be tested whether the spec of the service is correct

Reply:
Done, thanks